### ISSUE TITLE
[Bug]: Two pages can have each other as the content master document

### ISSUE THREAD

**Issue by ThisIsJustARandomGuy** [pimcore/pimcore#12765]
### Expected behavior

When trying to set a document as a content master document, Pimcore should throw an exception if the inheritance chain ever lands on the current document again. Currently, an exception `You cannot use the current document as a master document, please choose a different one.` is raised by this piece of code in the `setContentMasterDocumentId` method in `models/Document/PageSnippet.php`:

```PHP
if ($contentMasterDocumentId && $contentMasterDocumentId == $this->getId()) {
    throw new \Exception('You cannot use the current document as a master document, please choose a different one.');
}
```

However, this does not prevent a longer chain where, for example, the content master document refers back to the current document. I'd like to add a second check that traverses the inheritance chain and tries to find `$this->getId()` in it. If found, it will raise an exception, similar to the already existing one. This should be able to work as a drop in replacement for the existing if-statement:

```PHP

// Prevent an endless loop where the master document chain points back to the current document.
// We must not find $this->getId() in the master document's inheritance chain.
if(!empty($contentMasterDocumentId)) {
    $maxDepth = 20;
    $currentDocument = Document::getById($contentMasterDocumentId);

    while ($currentDocument && $maxDepth-- > 0) {
        if ($currentDocument->getId() === $this->getId()) {
            throw new \Exception('This document is already part of the master document chain, please choose a different one.');
        }

        $currentDocument = $currentDocument->getContentMasterDocument();
    }
}
```

This is mostly the same as the existing check except for these factors:
* I use `$contentMasterDocument->getId()` for the comparison
* `===` instead of `==`
* I don't check if $currentDocument is an `instanceof self` but I don't think it can ever turn out to something that does not have the `getId()` method I call.

I don't know if any of this could lead to problems, so I refrained from creating a pull request for now. I'd be happy to create one but am a bit short on time for testing.

### Actual behavior

Pimcore does not prevent a longer chain than 1. An out of memory exception is raised because of a resulting endless loop in `PageSnippet::getEditable()`

### Steps to reproduce

1. Create two documents. Let's call them /first and /second
2. Use /second as the content master document for /first
3. Use /first as the content master document for /second
4. Save and reload the document
5. Opening the document should throw an OutOfMemory exception

The chain can be arbitrarily long: /first pointing to /second, /second pointing to /third and /third pointing back to /first, for example

**Comment by aryaantony92** [comment]
@ThisIsJustARandomGuy Thank you. I have adapted your solution and also added unit test for the same.

**Comment by dvesh3** [comment]
Fixed by #12834
